shotover-proxy/src/message/mod.rs

```diff
@@ -91,6 +114,7 @@ impl Message {
     /// Providing just the Frame results in better performance when only the Frame is available.
     pub fn from_frame(frame: Frame) -> Self {
```

Comment:
I think we need to take a ProtocolType here in order to properly set codec_state